### PR TITLE
Fix duplicated orders

### PR DESCRIPTION
**Pull request by rmarinleal**

### CODE REVIEW

File: pagantis.php [hunk 1]
```diff
@@ -254,6 +254,23 @@ public function checkEnvVariables()
...
     {
+        try {
+            $tableName = _DB_PREFIX_.'pagantis_order';
+            $sql = ("SHOW TABLES LIKE '$tableName'");
+            $results = Db::getInstance()->ExecuteS($sql);
+            if (is_array($results) && count($results) === 1) {
+                $query = "select COLUMN_TYPE FROM information_schema.COLUMNS where TABLE_NAME='$tableName' AND COLUMN_NAME='ps_order_id'";
+                $results = $results = Db::getInstance()->ExecuteS($query);
+                if (is_array($results) && count($results) === 0) {
+                    $sql = "ALTER TABLE $tableName ADD COLUMN ps_order_id VARCHAR(50) AFTER order_id";
```
Comment (pgarcess) [inline pagantis.php]: varchar(50) or varchar(60) like the file sql/install.sql?

File: controllers/front/payment.php [hunk 2]
```diff
@@ -255,8 +255,8 @@ public function postProcess()
             if ($order instanceof \Pagantis\OrdersApiClient\Model\Order) {
                 $url = $order->getActionUrls()->getForm();
                 $orderId = $order->getId();
-                $sql = "INSERT INTO `" . _DB_PREFIX_ . "pagantis_order` (`id`, `order_id`)
-                     VALUES ('$cart->id','$orderId') 
+                $sql = "INSERT INTO `" . _DB_PREFIX_ . "pagantis_order` (`id`, `order_id`, `ps_order_id`)
```
Comment (pgarcess) [inline controllers/front/payment.php]: The field ps_order_id accept null, so it is not necessary fill the field in this case

File: controllers/front/notify.php [hunk 3]
```diff
@@ -294,12 +286,39 @@ public function validateAmount()
...
+            }
+
+            // Double check
+            $tableName = _DB_PREFIX_ . 'pagantis_order';
+            $sql = ('select ps_order_id from `' . $tableName . '` where `id` = ' . $this->merchantOrderId
+                . ' and `order_id` = \'' . $this->pagantisOrderId . '\''
+                . ' and `ps_order_id` is not null');
+            $results = Db::getInstance()->ExecuteS($sql);
+            if (is_array($results) && count($results) === 1) {
+                throw new WrongStatusException('PS->record found in ' . $tableName . ': already_processed');
```
Comment (pgarcess) [inline controllers/front/notify.php]: could be good write the order_id in the log information

Comment (rmarinleal) [reply controllers/front/notify.php]: Good idea, done.

File: controllers/front/payment.php (hunk 2 above)
Comment (rmarinleal) [reply controllers/front/payment.php]: sure, deleted

File: pagantis.php (hunk 1 above)
Comment (rmarinleal) [reply pagantis.php]: ops, 60 to preserve integrity ;) changed

**Review by pgarcess — APPROVED** [review]

**Comment by rmarinleal** [comment]
Corrupted PR

**Pull request closed without merging**